src/core_plugins/kibana/public/dashboard/dashboard_config.js

```diff
@@ -0,0 +1,22 @@
+import { uiModules } from 'ui/modules';
+uiModules.get('kibana')
+  .provider('dashboardConfig', () => {
+    let hideWriteControls = false;
+
+    return {
+      /**
+       * Part of the exposed plugin API - do not remove without careful consideration.
+       * @type {boolean}
+       */
+      turnHideWriteControlsOn() {
+        hideWriteControls = true;
+      },
+      $get() {
+        return {
+          getHideWriteControls() {
```

Comment:
I like the `get` prefix because I think it makes it clear that it's not a setter.  e.g. you can't immediately tell if `shouldHide..` would take a boolean and set the value of shouldHide, or is getting the shouldHide variable.  Though I agree it reads nicer when it's `if (shouldHide)...`.

I suppose I wouldn't be opposed to calling it `getShouldHideWriteControls`, but I'm not sure the `should` really adds anything. 

I'd prefer not to hold up the PR on this but it might be worthwhile to create another issue to come to a team conclusion on the best way to name getters and then add it to the style guide.  I've had reviews in the past with @cjcenizal that touch on this issue as well, and I don't think we ever settled on something concrete.